src/Illuminate/Translation/Translator.php

```diff
@@ -149,6 +149,12 @@ protected function makeReplacements($line, array $replace)
         $replace = $this->sortReplacements($replace);
 
         foreach ($replace as $key => $value) {
+            if (strpos($line, ':'.strtoupper($key)) !== false) {
+                $line = str_replace(':'.strtoupper($key), strtoupper($value), $line);
+            } elseif (strpos($line, ':'.ucfirst($key)) !== false) {
+                $line = str_replace(':'.ucfirst($key), ucfirst($value), $line);
+            }
+
             $line = str_replace(':'.$key, $value, $line);
```

Comment:
I had realized the strpos() conditionals are pointless overhead (too much Whisky this evening perhaps!) when I saw your comment. That's a much better improvement!